synthesizer/process/src/stack/execute.rs

```diff
@@ -115,6 +115,10 @@ impl<N: Network> StackExecute<N> for Stack<N> {
                     Operand::BlockHeight => {
                         bail!("Illegal operation: cannot retrieve the block height in a closure scope")
                     }
+                    // If the operand is the network id, throw an error.
+                    Operand::NetworkID => {
+                        bail!("Illegal operation: cannot retrieve the network id in a closure scope")
```

Comment:
What is the reason why this shouldn't be available in execution?
Same for below.

Reply:
Discussed offline. Excluding in execution and evaluation does not prevent the desired use case as you can directly pass in a u32 and check it against the network in a finalize block.